consensus/src/transaction_filter/mod.rs

```diff
@@ -14,19 +15,18 @@ impl TransactionFilter {
         Self { filter }
     }
 
-    pub fn filter(
-        &self,
-        block_id: HashValue,
-        timestamp: u64,
-        txns: Vec<SignedTransaction>,
-    ) -> Vec<SignedTransaction> {
+    pub fn filter(&self, block_id: HashValue, timestamp: u64, txns: &mut Vec<SignedTransaction>) {
```

Comment:
up to you, but taking ownership and returning ownership might be easier to follow on the caller side (and is unrelated to your change)

Reply:
Done.